### PR TITLE
No longer allow adding a repository folder to a repository folder when the repository folder is deactivated

### PR DESCRIPTION
**Pull request by tinagerber**
Since the repository workflow allows adding repository folders when repository folders are deactivated, the repository folder incorrectly appears in the add menu.

For [CA-2175]

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- API change:
  - [x] Documentation is updated
  - [x] API Changelog entry (see [guide]
- Upgrade steps (changes in profile):
  - [x] Make it deferrable if possible
  - [x] Execute as much as possible conditionally

[CA-2175]: https://4teamwork.atlassian.net/browse/CA-2175

### CODE REVIEW

File: opengever/repository/tests/test_repositoryfolder.py [hunk 1]
```diff
@@ -126,6 +127,16 @@ def test_only_repofolder_addable_when_already_contains_repositories(self, browse
             ['Repository Folder'],
             factoriesmenu.addable_types())
 
+    @browsing
+    def test_repofolder_not_addable_when_repofolder_is_deactivated(self, browser):
+        self.login(self.administrator, browser)
+        browser.open(self.branch_repofolder)
+        self.assertIn('Repository Folder', factoriesmenu.addable_types())
+        editbar.menu_option('Actions', 'Deactivate').click()
+        with self.assertRaises(ValueError) as cm:
```
Comment (deiferni) [inline opengever/repository/tests/test_repositoryfolder.py:136]: i think you cold do `assertFalse(factoriesmenu.visible())` to have a nicer test here.

File: opengever/core/profiles/default/workflows/opengever_repository_workflow/definition.xml [hunk 2]
```diff
@@ -236,8 +236,6 @@
     <permission-map
         name="opengever.repository: Add repositoryfolder"
         acquired="False">
-      <permission-role>Administrator</permission-role>
-      <permission-role>Manager</permission-role>
```
Comment (deiferni) [inline opengever/core/profiles/default/workflows/opengever_repository_workflow/definition.xml:240]: I would prefer to leave manager in. IMO we should always be privileged to bypass these restrictions for maintenance.

**Review by deiferni — COMMENTED** [review]
Two small suggestions. Apart from that 👍 .

File: opengever/core/profiles/default/workflows/opengever_repository_workflow/definition.xml (hunk 2 above)
Comment (tinagerber) [reply opengever/core/profiles/default/workflows/opengever_repository_workflow/definition.xml:240]: You think it is necessary that a manager can create new repository folders in deactivated repository folders? This is not the case with other content types, e.g. deactivated dossiers.  I think a manager knows that he has to reactivate the repository folder to create new repository folders.

Comment (deiferni) [reply opengever/core/profiles/default/workflows/opengever_repository_workflow/definition.xml:240]: ah yes, you're right. In that case it is good the way it is for consistency reasons. :shipit:!

**Review by deiferni — APPROVED** [review]

**Pull request merged**